crates/rollup-boost/src/server.rs

```diff
@@ -44,6 +44,9 @@ pub type Response = HttpResponse;
 pub type BufferedRequest = http::Request<Full<bytes::Bytes>>;
 pub type BufferedResponse = http::Response<Full<bytes::Bytes>>;
 
+// Returns (payload_option, builder_api_failed)
+pub type BuilderResult = Result<(Option<OpExecutionPayloadEnvelope>, bool), ErrorObject<'static>>;
```

Comment:
could you make Option<OpExecutionPayloadEnvelope>, bool into a struct, it would improve readability and easier to document what the fields mean